fastlane/lib/fastlane/actions/set_info_plist_value.rb

```diff
@@ -5,12 +5,19 @@ module SharedValues
 
     class SetInfoPlistValueAction < Action
       def self.run(params)
-        require "plist"
+        require 'plist'
 
         begin
           path = File.expand_path(params[:path])
           plist = Plist.parse_xml(path)
-          if params[:subkey]
+          if params[:map]
```

Comment:
This if statement is getting a bit unruly. Could we break some of these things out into their own methods to not have so much nested code?